stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1447 @@
+                                                   the enum and map the values over to the requests object"""
+                                                             .format(str(auth_schematic)))
+
+            except DataError as e:
+                self._common.print_type_error("""The authentication supplied {0} does not match the required
+                                                schema {1}. You can view the layout of the schema on the
+                                                `HTTPAuthenticationType` enum.
+
+                                                The error provided by the execution
+                                                {2}"""
+                                              .format(str(auth_details), str(auth_schematic), e))
+            except TypeError as e:
+                self._common.print_type_error("""The authentication details object passed to this function failed as
+                                                the type of this object is incorrect. The type passed down was
+                                                {0} and the expected type is a iterable value that matches the
+                                                `HTTPAuthenticationType` enum
+
+                                                The error provided by the execution
+                                                {1}"""
+                                              .format(type(auth_details), e))
+
+        else:
+            self._common.print_type_error("""The `auth_schematic` variable passed to the `set_auth` function"
+                                            is currently invalid. You need to pass down a schematic object from the
+                                            `HTTPAuthenticationType` Enum or a type" error will occur.
+                                            The current schematic passed to this function is: {0}
+                                            """.format(str(auth_schematic)))
+
+    def apply_mock(self, method, url, status_code=None, response=None):
+        adapter = requests_mock.Adapter()
+        self.mount_adapter(adapter)
+        adapter.register_uri(method, url,
+                             status_code=status_code,
+                             json=response)
+
+
+"""
+    The HTTP Helper Connection Handler is used to control the state of the connection outside of the Session() and
+    Request() object state.
+    Anything that can manipulate, create or fetch the state of the session should be contained within,
+    Functionality:
+        - Create and maintain the connection values outside of the Session() and Request() objects
+        - SET && GET HTTP Timeout
+        - SET && GET HTTP Retry Policy
+        - SET && GET HTTP SSL Verification
+        - SET && GET HTTP Proxy
+        - Sending the HTTP Request and Session to the Endpoint
+"""
+
+
+class HTTPHelperConnectionHandler:
+    # Private objects to keep internal state
```

Comment:
You are defining class variables here, not creating private properties.

Reply:
I'm referring to the _ variables inside the class.
Redundant comment will remove them